internal/lvmd/lvservice.go

```diff
@@ -181,20 +148,12 @@ func (s *lvService) RemoveLV(ctx context.Context, req *proto.RemoveLVRequest) (*
 
 func (s *lvService) CreateLVSnapshot(ctx context.Context, req *proto.CreateLVSnapshotRequest) (*proto.CreateLVSnapshotResponse, error) {
 	logger := log.FromContext(ctx).WithValues("name", req.GetName())
-
-	var snapType string
 	dc, err := s.dcmapper.DeviceClass(req.DeviceClass)
 	if err != nil {
 		return nil, status.Errorf(codes.NotFound, "%s: %s", err.Error(), req.DeviceClass)
 	}
-
-	switch dc.Type {
-	case lvmdTypes.TypeThin:
-		snapType = "thin-snapshot"
-	case lvmdTypes.TypeThick:
+	if dc.Type != lvmdTypes.TypeThin {
```

Comment:
I’d like to leave this code as-is, because:

- Changing it would stray from the main purpose of this PR.
  - At least it should be changed in another commit.
- A few years ago, we planned to support snapshots for thick volumes; when that feature is implemented (in the distant future), we can use the current switch-case code.